sdk/src/logs/simple_log_processor.cc

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/sdk/logs/simple_log_processor.h"
+
+#include <chrono>
+#include <vector>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+/**
+ * Initialize a simple log processor.
+ * @param exporter the configured exporter where log records are sent
+ */
+SimpleLogProcessor::SimpleLogProcessor(std::unique_ptr<LogExporter> &&exporter)
+    : exporter_(std::move(exporter))
+{}
+
+/**
+ * Batches the log record it receives in a batch of 1 and immediately sends it
+ * to the configured exporter
+ */
+void SimpleLogProcessor::OnReceive(
+    std::unique_ptr<opentelemetry::logs::LogRecord> &&record) noexcept
+{
+  std::vector<std::unique_ptr<opentelemetry::logs::LogRecord>> batch;
+  batch.emplace_back(std::move(record));
+
+  // Get lock to ensure Export() is never called concurrently
+  std::lock_guard<opentelemetry::common::SpinLockMutex> locked(lock_);
+
+  if (exporter_->Export(batch) != ExportResult::kSuccess)
+  {
+    /* TODO: alert user of the failed or timedout export result */
+  }
+}
+/**
+ *  The simple processor does not have any log records to flush so this method is not used
+ */
+bool SimpleLogProcessor::ForceFlush(std::chrono::microseconds timeout) noexcept
+{
+  return true;
+}
+
+/**
+ * TODO: This method should not block indefinitely. Should abort within timeout.
```

Comment:
You can remove this TODO. Timeouts should be handled by the exporter.

Reply:
My original concern was whether it was safe to make a user who writes their own exporter be responsible for this. I think if the processor `Shutdown` function didn't abort as a safety check, there _could_ be hanging - let me know whether you think this makes sense or not.

Reply:
Done